libs/vkd3d/resource.c

```diff
@@ -2215,6 +2215,13 @@ HRESULT d3d12_resource_validate_desc(const D3D12_RESOURCE_DESC1 *desc, struct d3
             }
             /* Fall through. */
         case D3D12_RESOURCE_DIMENSION_TEXTURE2D:
+            if (desc->SampleDesc.Count > 1 &&
+                    !(desc->Flags & (D3D12_RESOURCE_FLAG_ALLOW_DEPTH_STENCIL | D3D12_RESOURCE_FLAG_ALLOW_RENDER_TARGET)))
+            {
+                WARN("Multi-sampled textures must be created with render target or depth attachment usage.\n");
+                return E_INVALIDARG;
+            }
+            /* Fall through */
```

Comment:
Theis whole sample count thing seems weird, we're implicitly allowing multisampled 1D + 3D textures for some reason. Is this intended?

We might also want to error out if ALLOW_UAV is set with >1 sample and the device feature is not supported.

Reply:
Right, that was not intended.

Reply:
Fixed.